### ISSUE TITLE
Datepicker SCSS issue

### ISSUE THREAD

**Issue by joshpencheon** [PublicHealthEngland/ndr_ui#21]
There is an `@import` directive in the date picker SCSS that is compiling to a raw CSS `@import` because of the explicit `.css` extension on the path (i.e. it is not inlining the contents of the file during compilation).

Without the `.css` extension, most SCSS preprocessors would correctly still find it, but for Ruby this is still stalled in development: https://github.com/sass/sass/issues/556#issuecomment-397771726

**Comment by joshpencheon** [comment]
Adding the following:

```diff
diff --git a/app/assets/stylesheets/ndr_ui/datepicker.scss b/app/assets/stylesheets/ndr_ui/datepicker.scss
index cef36b8..1be30fe 100644
--- a/app/assets/stylesheets/ndr_ui/datepicker.scss
+++ b/app/assets/stylesheets/ndr_ui/datepicker.scss
@@ -1,4 +1,6 @@
-@import "bootstrap-datepicker-1.6.0-dist/css/bootstrap-datepicker3.css";
+// Reference a file that we've renamed to be .[s]css, to prevent a pass-through
+// of the @import as a CSS3 directive.
+@import "bootstrap-datepicker-1.6.0-dist/css/bootstrap-datepicker3";

 // Ensure date picker renders over the static nav
 .datepicker {
diff --git a/vendor/assets/bootstrap-datepicker-1.6.0-dist/css/bootstrap-datepicker3.css b/vendor/assets/bootstrap-datepicker-1.6.0-dist/css/bootstrap-datepicker3.scss
similarity index 100%
rename from vendor/assets/bootstrap-datepicker-1.6.0-dist/css/bootstrap-datepicker3.css
rename to vendor/assets/bootstrap-datepicker-1.6.0-dist/css/bootstrap-datepicker3.scss
```

...makes the error go away, and I _think_ means we don't need to worry about asset precompilation (assuming users are following the README and adding the relevant `require` directives). But I'd appreciate a second pair of eyes / someone playing before I cut a new release - @miles-smith ?

**Comment by joshpencheon** [comment]
(see https://vanseodesign.com/css/sass-the-import-directive/ for CSS vs. SCSS `@import`)

**Comment by miles-smith** [comment]
On the face of it, this sounds like it could be a quick/easy solution to the problem. I'll find some time to experiment with it and see if anything falls out.

**Comment by miles-smith** [comment]
Thinking about it, isn't removing the `.css` extension from the `@import` all that's needed? The appropriate `require` directives in the host app should be sufficient to ensure everything gets included during pre-compilation, unless I've missed something obvious?

**Comment by joshpencheon** [comment]
That would effectively be reverting f590f4e3007033 (which may not be the wrong answer)

I was trying to get test coverage of this last night - I've pushed the basic changes to the `issue_21_debug` - but was struggling to get the errors to manifest in the same was as they do in our internal application.

Whatever we do, it'd be good to have some test coverage of it!

**Comment by miles-smith** [comment]
True, and agreed.

Anecdotally, I'm not seeing any further issues on the project I've been playing with, but that's hardly conclusive evidence of a solid solution. I'll see what I can do with the engine's test app tomorrow.

**Comment by joshpencheon** [comment]
If I remove the extension, I get the following:

```
test/dummy master $ RAILS_ENV=test bundle exec rake assets:precompile
rake aborted!
Sass::SyntaxError: File to import not found or unreadable: bootstrap-datepicker-1.6.0-dist/css/bootstrap-datepicker3.
[...]
```

...as mentioned in the `sass/sass` link above, the Ruby library doesn't (currently) support look for CSS files unless the extension is explicitly used.

**Comment by miles-smith** [comment]
Ok, I think I've got some progress on this. I've pushed some changes onto the branch.

* Added symlink `bootstrap-datepicker.scss` that points to `bootstrap-datepicker3.css`
* Changed the `@import` to explicitly use the symlink. This ensures that the vendored CSS gets inlined correctly rather than generating an `@import url('bootstrap-datepicker3.css')` rule in the compiled CSS.

* Added a test which should fail/error in the event of unresolvable assets. It isn't an exact replication of the scenario we're seeing out in the wild but I think it should be enough to smoke out problems.
